joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/passes/ast/TypeDeclAstCreationPassTest.scala

```diff
@@ -31,6 +31,19 @@ class TypeDeclAstCreationPassTest extends RubyCode2CpgFixture {
       myClass.fullName shouldBe "Test0.rb::program.MyClass"
     }
 
+    // TODO: Need to be fixed.
+    "populate class name correctly for a derived class" in {
+      val cpg = code("""
+          |module ApplicationCable
+          |  class Channel < ActionCable::Channel::Base
+          |  end
+          |end
+          |""".stripMargin)
+      val List(myClass) = cpg.typeDecl.nameExact("Channel").l
+      myClass.name shouldBe "Channel"
+      myClass.fullName shouldBe "Test0.rb::program.ApplicationCable.ActionCable.Channel"
```

Comment:
@tuxology I think full name also should be `Test0.rb::program.ApplicationCable.Channel` instead of `Test0.rb::program.ApplicationCable.ActionCable.Channel`

Reply:
@pandurangpatil @fabsx00 I removed the base class from the names, but I don't know how how this impacts type hierarchy

Reply:
@tuxology basename doesn't become part of the fullname of the class at least in the contest of Java.

Reply:
As per my understanding - For a given class its `name` and `fullName` should only correspond to self. If it extends another class, the base class fullName should be present in `inheritsFromFullName`